Extensions/addressbar.js

```diff
@@ -0,0 +1,78 @@
+//* TITLE Addressbar Updater **//
+//* VERSION 1.0.1 **//
+//* DESCRIPTION Keeps the url in its place **//
+//* DEVELOPER srsutherland **//
+//* DETAILS Sets the url to the current post on your dashboard (to keep your place during tab crashes and refreshes). **//
+//* FRAME false **//
+//* BETA false **//
+//* SLOW false **//
+
+XKit.extensions.addressbar = new Object({
+
+	running: false,
+	intervalId: 0,
+	initialScroll: null,
+	
+	preferences: {
+		"focus_only": {
+			text: "Only run the extension when the page has focus",
+			default: true,
+			value: true
+		},
+	},
+
+	run: function() {
+		this.running = true;
+
+		if (XKit.interface.where().dashboard !== true || XKit.interface.where().endless !== true) {return; } //only useful on endless dashboard
+
+		XKit.extensions.addressbar.intervalId = setInterval(
+			XKit.extensions.addressbar.update_address, 500);
+
+		XKit.extensions.addressbar.initialScroll = history.scrollRestoration;
+		history.scrollRestoration = "manual";
+	},
+	
+	find_top: function() {
+		var offset = 200; //place slightly below navbar to ignore reblog controls
+		var scrollPos = $(window).scrollTop() + offset;
+		var minDist = Number.MAX_SAFE_INTEGER;
+		var id = null; //check later and do nothing to history if no post qualifies
+	
+		$(".posts .post").not("#new_post").not("#tumblr_radar").not(".new_post_buttons").each(function() {
```

Comment:
wouldn't this affect peepr too? i feel like that can't be intentional since peepr scrolling is done within its own element. assuming it's not, you can shorten this down to only affect real posts by using an attribute selector:
```js
$("[data-pageable] .post").each(function() {
```
(`data-pageable` is a data attribute only found on dashboard post containers, and using it like this targets any element with it, regardless of the value)